server/routes/add-steps/AddStepsController.ts

```
@@ -23,6 +26,21 @@ export default class AddStepsController {
       const planUuid = req.services.sessionService.getPlanUUID()
       const plan = await req.services.planService.getPlanByUuid(planUuid)
 
+      let assessmentDetailsForArea = null
+      const criminogenicNeedsData = req.services.sessionService.getCriminogenicNeeds()
+
+      // get assessment data
+      try {
+        const assessmentResponse = await req.services.assessmentService.getAssessmentByUuid(planUuid)
+        const assessmentData = assessmentResponse.sanAssessmentData
+
+        // 2. get the assessment details for the selected area of need
+        const areaConfig: AssessmentAreaConfig = areaConfigs.find(config => config.area === goal.areaOfNeed.name)
+        assessmentDetailsForArea = getAssessmentDetailsForArea(criminogenicNeedsData, areaConfig, assessmentData)
+      } catch {
+        /* swallow the error, missing data is handled in the template */
```

Can I ask the purpose and justification of an empty catch block here? I think it may be a little concerning to rely on the template to handle the error expressly as the template should primarily display the error outcome.

---

The variable that this block sets is set to null beforehand because we output a specific matching message to the user when this set of data is not available. There's no alternative fallback position if the service we call isn't available, and individual request failures are logged within the functions called here and so would be superfluous.

The catch only exists so that if the service we call isn't available then sentence plan itself doesn't crash when the error is propagated back up the stack.

---

As part of addressing Jake's comment above I've clarified the comment in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/393/commits/7f467f893e3f9244d50ac8be87f2fcc31f41b5ab

---

I agree with @JakeGoodwin22 suggestion to avoid the `let` associated and appreciate the explanation :-)